routes/printing.py

```diff
@@ -84,8 +70,23 @@ def print_documents():
                     lectures=lectures)
             sqla.session.add(dep)
             db.accounting.log_deposit(dep, user, data['cash_box'])
+            if documents:
+                name = data['cover_text'].split(' ')[0]
+                for _ in config.print_documents(
+                    doc_paths=[document_path(doc.id) for doc in documents],
+                    cover_text=data['cover_text'],
+                    printer=data['printer'],
+                    user=user.username,
+                    usercode=config.PRINTER_USERCODES[data['cash_box']],
+                    job_title="Odie-Druck für {} [{} Seiten]".format(name, num_pages)):
+                        pass
         sqla.session.commit()
+        yield('accounting succeeded', '')
+    except NonConfidentialException as e:
+        sqla.session.rollback()
+        raise NonConfidentialException(str(e)) from e
     except Exception as e:
         # in case of network troubles, we've just printed a set of documents but screwed up accounting.
-        raise NonConfidentialException('Printing succeeded, but account logging failed. Exception: ' + str(e)) from e
+        sqla.session.rollback()
+        raise NonConfidentialException('accounting failed. Exception: ' + str(e)) from e
```

Comment:
Accounting failed? You don't know that! Since you're doing everything at once, the only bit of information you have here is "something went wrong"